app/src/androidTest/java/com/wire/android/ui/LoginScreenTest.kt

```diff
@@ -0,0 +1,84 @@
+package com.wire.android.ui
+
+import androidx.compose.animation.ExperimentalAnimationApi
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.test.assertHasClickAction
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.hasTestTag
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performTextClearance
+import androidx.compose.ui.test.performTextInput
+import com.wire.android.ui.authentication.login.LoginScreen
+import com.wire.android.ui.authentication.welcome.WelcomeScreen
+import com.wire.android.ui.authentication.welcome.WelcomeViewModel
+import com.wire.android.ui.theme.WireTheme
+import com.wire.android.utils.WorkManagerTestRule
+import com.wire.android.utils.getViewModel
+import com.wire.kalium.logic.configuration.ServerConfig
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import org.junit.Rule
+import org.junit.Test
+
+@OptIn(
+    ExperimentalMaterial3Api::class, ExperimentalAnimationApi::class,
+    ExperimentalComposeUiApi::class, ExperimentalMaterialApi::class
+)
+@HiltAndroidTest
+class LoginScreenTest {
+
+    @get:Rule(order = 0)
+    var hiltRule = HiltAndroidRule(this)
+
+    // Second, as we are using a WorkManager
+    // In an instrumented test we need to ensure this gets initialized before launching any Compose/Activity Rule
+    @get:Rule(order = 1)
+    var workManagerTestRule = WorkManagerTestRule()
+
+    // Third, we create the compose rule using an AndroidComposeRule, as we are depending on instrumented environment ie: Hilt, WorkManager
+    @get:Rule(order = 2)
+    val composeTestRule = createAndroidComposeRule<WireActivity>()
+
+    @Test
+    fun iSeeLoginScreen() {
+        hiltRule.inject()
+
+        // Start the app
+        composeTestRule.setContent {
+            WireTheme {
+                LoginScreen(serverConfig = ServerConfig.DEFAULT)
+            }
+        }
+
+        composeTestRule.onNode(hasTestTag("emailField"), useUnmergedTree = true).assertIsDisplayed()
+        composeTestRule.onNode(hasTestTag("emailField")).performTextClearance()
```

Comment:
It's not passing for me, I think we should add `semantics` modifier to `WireTextField` because it has multiple text-related nodes thus it doesn't know which one to choose for this test, at least for setting the text - something like:
```
    Column(modifier = modifier.semantics(mergeDescendants = true) {
        setText {
            onValueChange(TextFieldValue(it.text, TextRange(it.text.length)))
            true
        }
    }) {
        if (labelText != null)
            Label(labelText, labelMandatoryIcon, state, interactionSource, colors)
        BasicTextField(
        ...
```

Reply:
They work now, for later I will be thinking about semantics to choose modifier. I will create another PR for those changes